### PR TITLE
{bio}[foss/2016b] RELION 2.0.3 w/ CUDA (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 3 out of 7 (1 easyconfigs in this PR)
ca014.camp.thecrick.org - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/152414ec1fe40d6aabc8842adfb966c7 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 4 (1 easyconfigs in this PR)
ca014.camp.thecrick.org - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/cccf4878b2589021747ea408279a67d9 for a full test report.

**Comment by boegel** [comment]
@verdurin What's going on with the failing tests? Did you accidentally remove the required `bzip2` module?

**Comment by verdurin** [comment]
@boegel no, this is the same problem I've referred to elsewhere.

The module does load interactively, but for some reason not when invoked this way (for me).

File: easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ pronounce rely-on) is a stand-alone computer program that employs
+ an empirical Bayesian approach to refinement of (multiple)
+ 3D reconstructions or 2D class averages in electron
+ cryo-microscopy (cryo-EM)."""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+toolchainopts = {'opt': True}
+
+source_urls = ['https://github.com/3dem/relion/archive/']
+sources = ['%(version)s.tar.gz']
```
Comment (verdurin) [inline easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb]: Add a checksum here, please.

**Comment by boegel** [comment]
@verdurin I assume you're saying loading the `bzip2` module directly works? Does `module load Mako/1.0.4-foss-2016b-Python-2.7.12` work?

File: easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+builddependencies = [
+    ('CMake', '3.6.2'),
+]
+
+dependencies = [
+    (cuda, cuver, '', True),
+    ('X11', '20160819'),
+    ('FLTK', '1.3.3'),
+]
```
Comment (verdurin) [inline easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb]: Sanity check?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2695.swalot.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/0e0688403f81c866fb023f9384c54873 for a full test report.

**Comment by verdurin** [comment]
@boegel - yes, that works too

**Comment by boegel** [comment]
@verdurin In the same environment, i.e. with `$MODULEPATH` set to `/camp/apps/eb/review/modules/all:/camp/apps/misc/modules:/camp/apps/eb/modules/all:/etc/modulefiles:/usr/share/modulefiles:/usr/share/Modules/modulefiles:/camp/apps/eb/intel-2017a/modules/all`?

**Comment by verdurin** [comment]
@boegel, yes, both load manually with that `$MODULEPATH`

**Comment by boegel** [comment]
@verdurin What about if you use the same explicit command `/usr/share/lmod/lmod/libexec/lmod python load Mako/1.0.4-foss-2016b-Python-2.7.12`, same problem or does it work?

I'm keen on figuring out what's going on here... :)

**Comment by verdurin** [comment]
@verdurin If I run that exact command, it spews out a load of logs and
doesn't load anything. 

Doesn't show an error, either.

See https://gist.github.com/verdurin/9ff6ce2a6a19d6fc75f05938d11927de

**Comment by boegel** [comment]
@verdurin That's expected, it's just spitting out the Python 'commands' to update the environment.

I'm running out of ideas here, not sure what's going on...

Maybe try running with `eb --debug-lmod` (although you may run into other problems, see https://github.com/easybuilders/easybuild-framework/issues/2094)

**Comment by boegel** [comment]
@hajgato Ping on remarks (SHA256 checksum + sanity check)?

File: easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb (hunk 1 above)
Comment (hajgato) [reply easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb]: @verdurin  Done

File: easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb (hunk 2 above)
Comment (hajgato) [reply easybuild/easyconfigs/r/RELION/RELION-2.0.3-foss-2016b-CUDA-8.0.61.eb]: @verdurin Done

**Comment by hajgato** [comment]
@boegel Sorry, I completely overlooked those requests.

**Comment by hajgato** [comment]
Ok, I am giving it up. I clearly do not have any ide how should I indent here.

**Comment by pforai** [comment]
One small remark here, RELION can be built with MPI support and we're running it with MPI support. Will try MPI plus CUDA once we have the HW setup.

**Comment by migueldiascosta** [comment]
@hajgato, this PR is being closed for the following reason(s): no activity for > 6 months, obsoleted by more recent PRs.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**